### PR TITLE
test: fix test_system_backup_and_restore_volume_with_backingimage

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/10052

#### What this PR does / why we need it:

fix test_system_backup_and_restore_volume_with_backingimage

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved timing and sequence handling in the system backup and restore test case.

- **New Features**
	- Introduced a new method to wait for volume creation during system restore testing.

### CODE REVIEW

**Comment by yangchiu** [comment]
Test results: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7964/

File: manager/integration/tests/test_system_backup_restore.py [hunk 1]
```diff
@@ -199,7 +200,7 @@ def test_system_backup_and_restore_volume_with_backingimage(client, core_api, vo
     check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 3, "ready")
 
     restored_volume = client.by_id_volume(volume_name)
-    wait_for_volume_restoration_completed(client, volume_name)
+    wait_for_volume_creation(client, volume_name)
```
Comment (mantissahz) [inline manager/integration/tests/test_system_backup_restore.py:203]: Does it mean we don't know whether the volume is restored?
Cause we don't check if the data is correct after restoring the system backup.

Comment (yangchiu) [reply manager/integration/tests/test_system_backup_restore.py:203]: Yes, you're right. The test case is missing a data integrity check. I'll create a separate ticket for that. Thank you!

Comment (yangchiu) [reply manager/integration/tests/test_system_backup_restore.py:203]: Using https://github.com/longhorn/longhorn/issues/10052 to track.

**Pull request closed without merging**